kapitan/targets.py

```diff
@@ -324,6 +340,46 @@ def compile_target(target_obj, search_paths, compile_path, ref_controller, **kwa
     logger.info("Compiled %s (%.2fs)", target_name, time.time() - start)
 
 
+def get_dependencies_per_target(target_objs):
+    """returns a mapping from target name to a list of dependency objects"""
+    mapping = {}
+    for target_obj in target_objs:
+        try:
+            dependencies = target_obj["dependencies"]
+            target_name = target_obj["vars"]["target"]
+            mapping[target_name] = []
+            for item in dependencies:
+                dependency_type = item["type"]
+                source_uri = item["source"]
+                output_path = item["output_path"]
+                if dependency_type == "git":
+                    del item["output_path"]
+                    mapping[target_name].append(Git(source_uri, output_path, **item))
+                elif dependency_type in ("http", "https"):
+                    mapping[target_name].append(Http(source_uri, output_path))
+                else:
+                    logger.error("type {} is not supported as dependency".format(dependency_type))
+                    assert False
+
+        except KeyError:
```

Comment:
Add logger.debug() messages in all exceptions to help during debugging :)

Reply:
Got it! Also, by right this error will not happen because of the json validation schema that limits type to one of http, https or git. Let me know if assert should at all be used.

Reply:
You could just thrown an exception instead of `assert False` and in that exception use a good descriptive message like the one you have in `logger.error("type {} is not supported as dependency".format(dependency_type))`

Reply:
Please raise KapitanError/CompileError or even create a new error from errors.py